### PR TITLE
Address some mypy type hinting issues

### PR DESCRIPTION
**Pull request by AitElBadaoui**
## Summary of Changes

This merge request addresses some of the type hinting issues identified by MyPy.
Ref: https://github.com/Quantum-Accelerators/quacc/issues/782.

### Requirements

- [X] My PR is focused on a [single feature addition or bugfix](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/best-practices-for-pull-requests#write-small-prs).
- [ ] My PR has relevant, comprehensive [unit tests](https://quantum-accelerators.github.io/quacc/dev/contributing.html#unit-tests).
- [X] My PR is on a [custom branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository) (i.e. is _not_ named `main`).

### Notes

- Before contributing for the first time, read the ["Guidelines" section](https://quantum-accelerators.github.io/quacc/dev/contributing.html#guidelines).
- If you are an external contributor, you will see a comment from [@buildbot-princeton](https://github.com/buildbot-princeton). This is solely for the maintainers.
- When your code is ready for review, ping one of the [active maintainers](https://quantum-accelerators.github.io/quacc/about/contributors.html#active-maintainers).

### CODE REVIEW

**Comment by buildbot-princeton** [comment]
Can one of the admins verify this patch?

**Comment by AitElBadaoui** [comment]
Hello @Andrew-S-RosenThe,
The missing line in the coverage is because none of the jobs call the run_and_summarize function with a None value in option_defaults. I'm not sure if it's worth it to mock a job just to cover this one line of code. Please let me know what you think.

To resolve the remaining MyPy issues, I need to understand the objective of this MR: https://github.com/Quantum-Accelerators/quacc/pull/1903. The changes introduced have led to multiple MyPy errors due to assigning tuples as default values to variables of different data types. Is _DEFAULT_SETTING meant to always be empty ?

Thank you !

**Comment by Andrew-S-Rosen** [comment]
Hi @AitElBadaoui! Thank you so much for putting together this PR. I really appreciate it!! These all look like great changes and, in fact, may even reduce some bugs.

As for the test coverage, don't worry about it for this PR. No need to worry about test coverage for type hinting changes here.

As for PR #1903, I can see how that might introduce some problems in hindsight. The reasoning behind the PR was that I thought `_DEFAULT_SETTING` (rather than `None`) would make it clearer to the end-user that a default setting was being used (quacc has a set of [default settings](https://quantum-accelerators.github.io/quacc/user/settings/settings.html) it imports at runtime). However, I see the problem with the type hinting. I chose to use `()` as a placeholder because `()` was simple and immutable. I also thought this approach would be nice because then `None` could actually mean something distinct from using the defaults. But the problem is that something like `qchem_cmd: str = _DEFAULT_SETTING` is clearly going to be flagged as a problem by `mypy` since `_DEFAULT_SETTING` is a tuple (even though the actual setting being used at runtime is indeed a `str`).

I guess since the type hint is now "wrong", would you recommend that I switch back to `None` in those cases? I can do that in a separate PR if you think that is the best approach.

File: src/quacc/runners/prep.py [hunk 1]
```diff
@@ -138,7 +138,7 @@ def calc_cleanup(
         symlink_path.unlink(missing_ok=True)
 
 
-def terminate(tmpdir: Path | str, exception: Exception) -> Exception:
+def terminate(tmpdir: Path, exception: Exception) -> None:
```
Comment (Andrew-S-Rosen) [inline src/quacc/runners/prep.py:141]: For this one, could you update the docstring to be

```
Returns
-------
None
```

Comment (Andrew-S-Rosen) [inline src/quacc/runners/prep.py:141]: It'd be nice to also support `Path | str` for `tmpdir`. Maybe the better option is to leave the original type hint for that and instead do `tmpdir = Path(tmpdir)` in the function itself?

**Review by Andrew-S-Rosen — COMMENTED** [review]
As for a review, here are my (minor) comments. Otherwise, this is in great shape for merging!

File: src/quacc/runners/prep.py (hunk 1 above)
Comment (AitElBadaoui) [reply src/quacc/runners/prep.py]: That sounds good, I'll do that.

**Comment by AitElBadaoui** [comment]
Yes, I agree that we should switch back to None. After that, I can create a new PR to address other mypy issues. 
Thank you for the review!

**Comment by Andrew-S-Rosen** [comment]
@AitElBadaoui: wonderful! I will take are of that and merge this PR when I'm back at my computer in an hour or so. 👍

**Comment by Andrew-S-Rosen** [comment]
Merging this one. Thanks!

**Pull request merged**